src/renderer/utils/sync-box/provide-initial-values-for-sync-boxes.injectable.ts

```diff
@@ -7,21 +7,35 @@ import { beforeFrameStartsInjectionToken } from "../../before-frame-starts/befor
 import syncBoxInitialValueChannelInjectable from "../../../common/utils/sync-box/sync-box-initial-value-channel.injectable";
 import createSyncBoxStateInjectable from "../../../common/utils/sync-box/sync-box-state.injectable";
 import { requestFromChannelInjectionToken } from "../../../common/utils/channel/request-from-channel-injection-token";
+import { runInAction } from "mobx";
+import type { SyncBox } from "../../../common/utils/sync-box/sync-box-injection-token";
+import { syncBoxInjectionToken } from "../../../common/utils/sync-box/sync-box-injection-token";
+import assert from "assert";
 
 const provideInitialValuesForSyncBoxesInjectable = getInjectable({
   id: "provide-initial-values-for-sync-boxes",
 
   instantiate: (di) => {
     const requestFromChannel = di.inject(requestFromChannelInjectionToken);
     const syncBoxInitialValueChannel = di.inject(syncBoxInitialValueChannelInjectable);
-    const setSyncBoxState = (id: string, state: any) => di.inject(createSyncBoxStateInjectable, id).set(state);
+
+    const syncBoxes = di.injectMany(syncBoxInjectionToken);
+
+    const setSyncBoxState = (syncBox: SyncBox<any>, state: any) =>
+      di.inject(createSyncBoxStateInjectable, syncBox.id).set(state);
 
     return {
       run: async () => {
         const initialValues = await requestFromChannel(syncBoxInitialValueChannel);
 
-        initialValues.forEach(({ id, value }) => {
-          setSyncBoxState(id, value);
+        runInAction(() => {
+          initialValues.forEach(({ id, value }) => {
+            const syncBox = syncBoxes.find((box) => box.id === id);
+
+            assert(syncBox);
+
+            setSyncBoxState(syncBox, value);
+          });
```

Comment:
Note: This change here would be covered by that skipped unit test having TODO. Importance of that rises because this is something that we would have seen.